### PR TITLE
allows response to have no body yet still be parseable

### PR DESCRIPTION
**Pull request by Uepsilon**

### CODE REVIEW

**Comment by Uepsilon** [comment]
this will handle #20

**Comment by Plsr** [comment]
Unfortunately, I don't have too much time to look into this right now, but this feels kind of sketchy to me, catching an error based on a string in the error message and then just returning anyways. 

Wouldn't it be possible to just check if there is something to parse at all and just not parse if the body is empty? Maybe something like [this](https://stackoverflow.com/a/37318584/4181679)?

**Comment by Uepsilon** [comment]
will add the following things:
- add request header `Accept: application/json`
- add parsing for `Content-Type: application/json` before actually parsing json 
- skip parsing for json if `Content-Type` not `application/json`

**Comment by Plsr** [comment]
Sounds awesome 👍

**Comment by Uepsilon** [comment]
Some clients actually mark the response as `application/json` but don't attach a body. in case we only skip when not `application/json`, this would still explode for those cases. added a more elegant and more general parsing note and some loggin

File: src/index.js [hunk 1]
```diff
@@ -83,26 +88,33 @@ export default class Snuffles {
...
+        const headers = {}
+        res.headers.forEach((value, key) => (headers[key] = value))
+
+        const resultBase = { status, headers, body: {} }
+
+        return res
+          .json()
+          .then(json => ({ ...resultBase, body: json }))
+          .catch(() => {
+            if (headers['Content-Cype'] === 'application/json') {
```
[the remark this answers is not in the document]
Comment (jhilden) [reply src/index.js]: see: https://github.com/railslove/snuffles/pull/22

**Comment by Plsr** [comment]
lgtm, lets wait until #22 is merged and then get this released together 👍

File: src/index.js [hunk 2]
```diff
@@ -83,20 +88,26 @@ export default class Snuffles {
...
-          status: res.status,
-          headers: res.headers,
-          body: json
-        }))
-      )
+      .then(res => {
+        const { status } = res
+        const resultBase = { status, headers: res.headers, body: {} }
+
+        const contentLength = res.headers.get('Content-Length')
```
Comment (timomeh) [inline src/index.js]: Ist das React Native/iOS Safari approved? Mit `headers` hatten wir ja davor schon Probleme.

**Comment by Uepsilon** [comment]
@timomeh please revise

**Pull request merged**